File: next-gen-residential-proxies/Ruby/proxy.rb
require_relative './settings'

def create_proxy_by_url(url)
  country = nil
  parsed_url = url

  url_parts = url.split ';'
  if url_parts.length == 2
    parsed_url = url_parts[0]
    country = url_parts[1]
  end

  [
    parsed_url,
    country,
    "http://#{get_setting(:Username)}:#{get_setting(:Password)}@#{get_setting(:ProxyAddress)}"
  ]
end

# func createProxyByUrl(url string) (string, string) {
#   urlParts := strings.Split(url, ";")
# if len(urlParts) == 2 {
#   return urlParts[0], fmt.Sprintf(
#     "http://customer-%s-cc-%s:%s@%s",
#     Username,
#     urlParts[1],
#     Password,
#     ProxyAddress,
#     )
# }
#
# return url, fmt.Sprintf(
#   "http://customer-%s:%s@%s",
#   Username,
#   Password,
#   ProxyAddress,
#   )
# }

File: next-gen-residential-proxies/Java/src/main/java/Proxy.java
public class Proxy {
    private final String url;
    private final String proxyHost;
    private final String proxyUsername;
    private final String proxyPassword;
    private final String country;

    public Proxy(String url,
                 String proxyHost,
                 String proxyUsername,
                 String proxyPassword,
                 String country) {
        this.url = url;
        this.proxyHost = proxyHost;
        this.proxyUsername = proxyUsername;
        this.proxyPassword = proxyPassword;
        this.country = country;
    }

    public String getProxyHost() {
        return proxyHost;
    }

    public String getProxyUsername() {
        return proxyUsername;
    }

    public String getProxyPassword() {
        return proxyPassword;
    }

    public String getUrl() {
        return url;
    }

    public String getCountry() {
        return country;
    }

    public static Proxy createProxyByUrl(String url) {
        var parsedUrl = url;
        String country = null;

        var urlParts = url.split(";");
        if (urlParts.length == 2) {
            parsedUrl = urlParts[0];
            country = urlParts[1];
        }

        return new Proxy(
                parsedUrl,
                Settings.PROXY_ADDRESS,
                Settings.USERNAME,
                Settings.PASSWORD,
                country
        );
    }
}
